core/services/ocr2/plugins/ocr2keeper/evm21/core/payload.go

```diff
@@ -0,0 +1,86 @@
+package core
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+// UpkeepTriggerID returns the identifier using the given upkeepID and trigger.
+// It follows the same logic as the contract, but performs it locally.
+func UpkeepTriggerID(id *big.Int, trigger []byte) string {
+	idBytes := id.Bytes()
+
+	combined := append(idBytes, trigger...)
+
+	triggerIDBytes := crypto.Keccak256(combined)
+	triggerID := hex.EncodeToString(triggerIDBytes)
```

Comment:
🏄 I would be inclined to just do `return hex.EncodeToString(triggerIDBytes)`

Reply:
This was just moved from other file :D Will make it more elegant